lib/honeybadger/config/defaults.rb

```diff
@@ -303,7 +303,10 @@ class Boolean; end
       },
       :'rails.subscriber_ignore_sources' => {
         description: "Sources (strings or regexes) that should be ignored when using the Rails' (7+) native error reporter.",
-        default: [],
+        # External libraries (eg Sidekiq, Resque) may wrap their execution in Rails' executor.
+        # But this means errors will first be reported by Rails.error, before the library's native error handler
+        # We ignore these reports, since the native error handler provides more context (such as job details)
+        default: ['application.active_support'],
```

Comment:
Since this is global config, what is the scenario where I'm in a puma process and an active support error happens? Is the Rails 7 default to report the error via the subscriber, but we're selectively ignoring that and reporting the error from our Rack middleware instead?

Reply:
Yes, if the Rack middleware is registered, it will report errors.

What this PR does is disable the Rails ErrorSubscriber for _unhandled_ errors on Rails < 7.1. Unhandled errors only, because we can safely ignore them, and our other integrations will catch them (Rack middleware, SIdekiq handler, etc). This way we can still catch errors with the right context, while still supporting manual calls of `Rails.error.handle`.

But now, going over the issues Sentry had with it, I'm thinking we should apply this change on 7.1+ as well. 🤔  Our integration handlers are older and more robust, and for unexpected errors, aht's really important.

Reply:
> But now, going over the issues Sentry had with it, I'm thinking we should apply this change on 7.1+ as well. 🤔 Our integration handlers are older and more robust, and for unexpected errors, aht's really important.

That makes sense to me.

Reply:
Updated. Now, we always defer unhandled errors to the integration's error handler.